HIP-6.md

```diff
@@ -0,0 +1,147 @@
+| hip | title                   | status | author | created    |
+| --- | ----------------------- | ------ | ------ | ---------- |
+| 6   | Validator announcements | Draft  | asaj   | 2022-12-21 |
+
+### **Brief Summary / Abstract**
+
+Defines an optional specification for Hyperlane validators to announce themselves and the location of their signed checkpoints.
+
+### **Motivation**
+
+Hyperlane relayers need to aggregate validator signed checkpoints in order to deliver interchain messages.
+
+At present, relayers are manually configured with a list of validators and the location of their signed checkpoints.
+
+Relying on manual configuration creates friction to the introduction of additional validators via sovereign consensus.
+
+This HIP defines a standard for validators to announce themselves to relayers via a smart contract on the underlying blockchain.
+
+Relayers should query this contract to discover the location of validator signed checkpoints.
+
+### **Tech Spec**
+
+#### Announcment attestations
+
+The validator announcement protocol relies on validators signing attestations with the following semantic meaning:
+
+> "Checkpoints for the mailbox contract with address _m_ on chain _d_, signed by the validator with address _v_, can be found at location _l_
+
+The Hyperlane validator binary should sign the following data upon startup.
+
+```
+/**
+ * @notice Returns the digest that Hyperlane validators should sign to support
+ * the announcement protocol
+ * @param _domain The origin domain of the Mailbox being validated
+ * @param _mailbox The address of the Mailbox being validated, as bytes32
+ * @param _storageType An enum representing the modality used to store
+ * signed checkpoints (e.g. AWS S3)
+ * @param _storageMetadata Information encoding the location of signed
+ * checkpoints, specific to the storage modaility (e.g. S3 bucket URL)
+ * @return The digest to EIP-191 sign
+ */
+function getDigestToSign(
+    uint32 _domain,
+    bytes32 _mailbox,
+    uint8 _storageType,
+    bytes calldata _storageMetadata
+)
+    external
+    pure
+    returns (bytes32)
+{
+    bytes32 _domainHash = keccak256(
+        abi.encodePacked(_domain, _mailbox, "HYPERLANE")
+    );
+    return keccak256(abi.encodePacked(_domainHash, _storageType, _storageMetadata));
+}
+```
+
+#### Storage types
+
+To be forward compatible with alternative storage modalities, the announcement protocol includes a "storage type" parameter.
+
+Upon adoption, the only supported storage type will be AWS S3, which has `storageType == 1`.
+
+For the S3 type, `_storageMetadata` should be the base64-encoded URL of the S3 bucket.
+
+Future storage types may be proposed in future HIPs. The content of `_storageMetadata` is expected to vary by type.
```

Comment:
This is more for my own personal curiosity, but what other storage types that are not URLs would you imagine?

Reply:
IPFS CID? But tbh I think that misses the point, we shouldn't have to try to guess when it's so easy to make it more general than a URL

Reply:
this feels like it fits in another HIP